fdbserver/ReadProxyServer.actor.cpp

```diff
@@ -0,0 +1,206 @@
+			GetKeyReply reply = wait(loadBalance(ssi.second, &StorageServerInterface::getKey,
+			                                     GetKeyRequest(keySel, req.version), TaskPriority::DefaultPromiseEndpoint, false,
+			                                     cx->enableLocalityLoadBalance ? &cx->queueModel : NULL));
+			req.reply.send(reply);
+			break;
+		} catch (Error& e) {
+			if (e.code() == error_code_wrong_shard_server || e.code() == error_code_all_alternatives_failed) {
+				cx->invalidateCache(keySel.getKey(), keySel.isBackward());
+
+				wait(delay(CLIENT_KNOBS->WRONG_SHARD_SERVER_DELAY, TaskPriority::DefaultEndpoint));
+			} else if (e.code() != error_code_actor_cancelled) {
+				req.reply.sendError(e);
+				break;
+			} else {
+				break;
+			}
+		}
+	}
+
+	return Void();
+}
+
+ACTOR Future<Void> getValue(GetValueRequest req, Database cx)
+{
+	state Key key = req.key;
+	state Version ver = req.version;
+
+	loop {
+		TraceEvent("VisheshReadValue");
+		state Optional<UID> getValueID = Optional<UID>();
+		state uint64_t startTime;
+		state double startTimeD;
+		try {
+			++cx->getValueSubmitted;
+			startTime = timer_int();
+			startTimeD = now();
+			++cx->transactionPhysicalReads;
+			if (CLIENT_BUGGIFY) {
+				throw deterministicRandom()->randomChoice(
+					std::vector<Error>{ transaction_too_old(), future_version() });
+			}
+
+			pair<KeyRange, Reference<LocationInfo>> ssi = wait( getKeyLocation(cx, key, &StorageServerInterface::getValue) );
+			GetValueReply reply = wait(loadBalance(ssi.second, &StorageServerInterface::getValue,
+												   GetValueRequest(key, ver, getValueID), TaskPriority::DefaultPromiseEndpoint, false,
+				                         cx->enableLocalityLoadBalance ? &cx->queueModel : NULL));
+
+			double latency = now() - startTimeD;
+			cx->readLatencies.addSample(latency);
+			cx->getValueCompleted->latency = timer_int() - startTime;
+			cx->getValueCompleted->log();
+
+			req.reply.send(reply.value);
+			return Void();
+		} catch (Error& e) {
+			TraceEvent("VisheshReadValueError").detail("Error", e.what()).detail("Key", req.key.toString());
+			cx->getValueCompleted->latency = timer_int() - startTime;
+			cx->getValueCompleted->log();
+			if (e.code() == error_code_wrong_shard_server || e.code() == error_code_all_alternatives_failed ||
+				(e.code() == error_code_transaction_too_old && ver == latestVersion) ) {
+				cx->invalidateCache( key );
+				wait(delay(CLIENT_KNOBS->WRONG_SHARD_SERVER_DELAY, TaskPriority::DefaultOnMainThread));
+			} else if (e.code() == error_code_actor_cancelled) {
+				req.reply.sendError(timed_out());
+				throw e;
+			} else {
+				req.reply.sendError(e);
+				break;
+			}
+		}
+	}
+
+	return Void();
+}
+
+ACTOR Future<Void> readProxyServerCore(ReadProxyInterface readProxy, Reference<AsyncVar<ServerDBInfo>> serverDBInfo) {
+	state Database cx = openDBOnServer(serverDBInfo, TaskPriority::DefaultEndpoint, true, true);
+	state ActorCollection actors(false);
```

Comment:
you are not waiting on actors.getResult(), so errors thrown by reads will not kill the readProxy